sqlglot/dialects/starrocks.py

```diff
@@ -81,4 +134,23 @@ class Generator(MySQL.Generator):
             exp.UnixToTime: rename_func("FROM_UNIXTIME"),
         }
 
+        def create_sql(self, e: exp.Create) -> str:
+            # Starrocks' primary key is defined outside of the schema, so we need to move it there
+            schema = e.this
+            if isinstance(schema, exp.Schema):
+                primary_key = schema.find(exp.PrimaryKey)
+                if primary_key:
+                    props = e.args.get("properties")
+                    if props:
+                        # Verify if the first one is an engine property. Is true then insert it after the engine,
+                        # otherwise insert it at the beginning
+                        if isinstance(props.expressions[0], exp.EngineProperty):
+                            props.expressions.insert(1, primary_key.pop())
+                        else:
+                            props.expressions.insert(0, primary_key.pop())
+                    else:
+                        e.set("properties", exp.Properties(expressions=[primary_key.pop()]))
```

Comment:
Can we have the `PRIMARY KEY` property at the end? If that's true then we can always just append and simplify this section. I think we should refactor it a bit regardless, e.g. one idea is to use `find` to search for an `EngineProperty` and then insert in its `.index` + 1 or something. Also if `properties` is not set you could do that beforehand so the section after it always inserts to an existing `properties` arg.

Reply:
No, key_desc always appears after engine type or first.

Reply:
Appreciate this suggestion, I will change it